test/functional/rpc_whitelist.py

```diff
@@ -92,5 +103,60 @@ def run_test(self):
         self.log.info("Strange test 5")
         assert_equal(200, rpccall(self.nodes[0], self.strange_users[4], "getblockcount").status)
 
+        self.test_rpcwhitelistdefault_0_no_permissions()
+        self.test_rpcwhitelistdefault_1_no_permissions()
+        self.test_rpcwhitelistdefault_1_with_user_permissions()
+
+    def test_rpcwhitelistdefault_0_no_permissions(self):
+        """
+        * rpcwhitelistdefault=0
+        * No Permissions defined
+        Expected result: * user3 (not whitelisted) can access any method
+        """
+        unrestricted_user = self.users[2]
+        for permission in self.COMMON_PERMISSIONS:
+            self.log.info("[" + unrestricted_user[0] + "]: Testing unrestricted user permission (" + permission + ")")
+            assert_equal(200, rpccall(self.nodes[0], unrestricted_user, permission).status)
+
+    def test_rpcwhitelistdefault_1_no_permissions(self):
+        """
+        * rpcwhitelistdefault=1
+        * No Permissions defined
+        Expected result: * user3 (not whitelisted) can not access any method
+        """
+        # Rewrite file configurations
```

Comment:
3123eea: I wonder if [`replace_in_config`](https://github.com/bitcoin/bitcoin/blob/c2d15d993ef06d97d4c117012bda6efa3dcbac45/test/functional/test_framework/test_node.py#L441) can be used here?

Reply:
@stratospher thanks for the suggestion , update [https://github.com/bitcoin/bitcoin/pull/29858/commits/fcf0ead6cd358fd35909e1102bdae2c176b0ace6](https://github.com/bitcoin/bitcoin/pull/29858/commits/fcf0ead6cd358fd35909e1102bdae2c176b0ace6)